tutor/templates/jobs/init/mounted-edx-platform.sh

```diff
@@ -0,0 +1,26 @@
+# When a new local copy of edx-platform is bind-mounted, certain build
+# artifacts from the openedx image's edx-platform directory are lost.
+# We regenerate them here.
+
+if [ -f /openedx/edx-platform/bindmount-canary ] ; then
+	# If this file exists, then edx-platform has not been bind-mounted,
+	# so no build artifacts need to be regenerated.
+	echo "Using edx-platform from image (not bind-mount)."
+	echo "No extra setup is required."
+	exit
+fi
+
+echo "Performing additional setup for bind-mounted edx-platform."
+set -x # Echo out executed lines
+
+# Regenerate Open_edX.egg-info
+pip install -e .
+
+# Regenerate node_modules
+npm clean-install
+
```

Comment:
There is an issue with `npm clean-install` i.e./aliased `npm ci`. 

The issue is the new version of npm >8.5.x would check depencey conflict and it would fail/errored if it exits which is the case until now in edx-platform.

I don't if it would work here, because I am not sure what the deafult version of `npm` in tutor. 

Anyway we have two option here: 
- stick to npm install as the de facto of tutor 
- pin it to spesfic version, similar to how to its done in edx-platform https://github.com/openedx/edx-platform/blob/75715e51feb49277db09027db5db398bb7046536/.github/workflows/js-tests.yml#L33

Lastly if this is expected to run **only** in dev mode, then I think in any case the default can `npm install` is okay, but for our project where devs probably might not want to touch any JS then `npm ci` might be better. 

Reference `npm ci` issue:
- a recent attemtp for me to resolve openedx/edx-platform/pull/31889 
- openedx/wg-frontend/issues/26
- openedx/wg-frontend/issues/105

Reply:
Ghassan, according to my tests `npm ci` works in olive.2. I recently opened a PR to replace `install` by `clean-install`: https://github.com/overhangio/tutor/pull/812/files
Why would it work in this PR, but not elsewhere?

Reply:
Yes you are right @regisb, it works because by deafault
- tutor install node 16.14.0 https://github.com/overhangio/tutor/blob/732e0c43be5b801b7c62892bdc9b44bc74b7263f/tutor/templates/build/openedx/Dockerfile#L115 
- node 16.14.0 comes preshipped with npm 8.3.1 [^1]
- And the issue will come up if you upgrade npm version to > 8.5 i.e manually running `npm i -g npm@latest` 
- Also it will come up if for node v18 upgrade, hence version 18 comes preshipped with npm 8.x hence [^2]

Lastly usually I see this pattern: 
- install a node version 
- install latest of npm `npm -g npm@latest` 
If tutor/upstream does that it would fail hence the pinning of npm verison in my above commet. 

In summary it's an upstream issue, it doesn't affect tutor so far, it might in the future if tutor to upgrade to node 18 while its not fixed upstream yet, or it might affect us if we want to follow the above pattern _which I thinnk its the recommend way secuirty wise_. But in any case this I guess that out of scope of this PR.  


[^1]: changelog of node 16.4.0 https://github.com/nodejs/node/blob/v16.14.0/doc/changelogs/CHANGELOG_V16.md#commits 
[^2]: changelog of node 18.0.0 https://github.com/nodejs/node/blob/v18.0.0/doc/changelogs/CHANGELOG_V18.md#commits